Content.Server/_NF/BountyContracts/BountyContractSystem.cs

```diff
@@ -0,0 +1,140 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Server.CartridgeLoader;
+using Content.Server.Chat.Systems;
+using Content.Server.GameTicking.Events;
+using Content.Server.StationRecords.Systems;
+using Content.Shared._NF.BountyContracts;
+using Content.Shared.Access.Systems;
+using Robust.Shared.Map;
+
+namespace Content.Server._NF.BountyContracts;
+
+/// <summary>
+///     Used to control all bounty contracts placed by players.
+/// </summary>
+public sealed partial class BountyContractSystem : SharedBountyContractSystem
+{
+    private ISawmill _sawmill = default!;
+
+    [Dependency] private readonly CartridgeLoaderSystem _cartridgeLoaderSystem = default!;
+    [Dependency] private readonly StationRecordsSystem _records = default!;
+    [Dependency] private readonly AccessReaderSystem _accessReader = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        _sawmill = Logger.GetSawmill("bounty.contracts");
+
+        SubscribeLocalEvent<RoundStartingEvent>(OnRoundStarting);
+        InitializeUi();
+    }
+
+    private void OnRoundStarting(RoundStartingEvent ev)
+    {
+        // use nullspace entity to store all information about contracts
+        var uid = Spawn(null, MapCoordinates.Nullspace);
```

Comment:
also if we are creating an entity on round start, we may want to have a round cleanup to remove any entities in nullspace that might still carry this comp through rounds in case a proper round restart doesn't finish properly.  I have seen other things in vanilla SS14 in the past that get caught persisting between rounds that shouldn't because the round restart fails for some other reason.  we can either cache and hang on to this entity and then remove it upon round restart cleanup event or you can do another entityquery on round restart cleanup and just delete any that have your component etc.

Reply:
I added entity query to delete any BountyData on round start and spawn a new one. I think in future iterations, I will move this to in game server. This will avoid this nullspace entity hack and also allow some funny stuff - like connecting to NT - Syndicate bounties servers.